metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MySqlParentChildRelMetaDataService.java

```diff
@@ -0,0 +1,229 @@
+            ps.setString(3, childName.toString());
+            ps.setString(4, childUUID);
+            ps.setString(5, type);
+            return ps;
+        });
+        log.info("Successfully create parent child relationship with parent={}, parentUUID={}, "
+                + "child={}, childUUID={}, relationType = {}",
+            parentName, parentUUID, childName, childUUID, type
+        );
+    }
+
+    @Override
+    public void deleteParentChildRelation(final QualifiedName parentName,
+                                          final String parentUUID,
+                                          final QualifiedName childName,
+                                          final String childUUID,
+                                          final String type) {
+        log.info("Deleting parent child relationship with parent={}, parentUUID={}, "
+                + "child={}, childUUID={}, relationType = {}",
+            parentName, parentUUID, childName, childUUID, type
+        );
+        jdbcTemplate.update(connection -> {
+            final PreparedStatement ps = connection.prepareStatement(SQL_DELETE_PARENT_CHILD_RELATIONS);
+            ps.setString(1, parentName.toString());
+            ps.setString(2, parentUUID);
+            ps.setString(3, childName.toString());
+            ps.setString(4, childUUID);
+            ps.setString(5, type);
+            return ps;
+        });
+        log.info("Successfully delete parent child relationship with parent={}, parentUUID={}, "
+                + "child={}, childUUID={}, relationType = {}",
+            parentName, parentUUID, childName, childUUID, type
+        );
+    }
+
+    @Override
+    public void rename(final QualifiedName oldName, final QualifiedName newName) {
+        try {
+            renameParent(oldName, newName);
+            renameChild(oldName, newName);
+            log.info("Successfully rename parent child relationship for oldName={}, newName={}",
+                oldName, newName
+            );
+        } catch (RuntimeException e) {
+            log.error("Failed to rename entity", e);
+            throw e;
+        }
+    }
+
+    private void renameParent(final QualifiedName oldName, final QualifiedName newName) {
+        jdbcTemplate.update(connection -> {
+            final PreparedStatement ps = connection.prepareStatement(SQL_RENAME_PARENT_ENTITY);
+            ps.setString(1, newName.toString());
+            ps.setString(2, oldName.toString());
+
+            return ps;
+        });
+    }
+
+    private void renameChild(final QualifiedName oldName, final QualifiedName newName) {
+        jdbcTemplate.update(connection -> {
+            final PreparedStatement ps = connection.prepareStatement(SQL_RENAME_CHILD_ENTITY);
+            ps.setString(1, newName.toString());
+            ps.setString(2, oldName.toString());
+            return ps;
+        });
+    }
+
+    @Override
+    public void drop(final QualifiedName name) {
```

Comment:
nit: we could combine into one sql (delete from ___ where parent = ? or child = ?), but its not necessarily a safer option since this class has a txn manager on it

Reply:
I agree but I decide to separate them out to anticipate in the future where we need to only drop one of them